packages/babel-helpers/src/helpers.js

```diff
@@ -588,28 +588,54 @@ helpers.interopRequireDefault = helper("7.0.0-beta.0")`
 `;
 
 helpers.interopRequireWildcard = helper("7.0.0-beta.0")`
+  function _createRequireWildcardCache(obj) {
+    if (typeof global._REQUIRE_WILD_CACHE === 'undefined') {
+      global._REQUIRE_WILD_CACHE = typeof WeakMap === 'function' ? new WeakMap() : null;
+    }
+
+    const cache = global._REQUIRE_WILD_CACHE;
```

Comment:
While `global` should always work in node, I would prefer to keep helpers self-contained.

I'd like to see this as:
```js
function _createRequireWildcardCache() {
  var map = typeof WeakMap !== "function" ? new WeakMap() : {
    has: function() { return false },
    set: function() {},
    get: function() { return null }
  };

  _createRequireWildcardCache = function() { return map };
  return map;
}

export default function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj;
  }

  var cache = _createRequireWildcardCache();
  if (cache.has(obj)) return cache.get(obj);

  // ...

  cache.set(obj, newObj);
  return newObj;
}
```

Since this disable the optimization when helpers are used inline, let's wait for @babel/babel's opinion.

Reply:
@nicolo-ribaudo Your approach also looks pretty cool! But, the reason for using `global` is to keep the cache even if `@babel/plugin-transform-runtime` is not available.

Please let me know if babel team have an internal rule that does not use `global` or any other better approach.

Reply:
Using `global` referenced is often causing i.e. webpack to include polyfills for it when bundling for web (at least on 1-3 versions, most likely on v4 too, not sure about v5 - as I've heard about some changes regarding this logic in there). Certainly something to consider before merging this in.

Reply:
#9232